spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClient.java

```diff
@@ -0,0 +1,681 @@
+            // Ignore the exception. The backfill succeeded but the update to
+            // the commit owner failed. The main purpose of a backfill operation is to ensure that
+            // UUID commit is physically copied to a standard commit file path. A failed update to
+            // the commit owner is not critical.
+        }
+    }
+
+    @Override
+    public Map<String, String> registerTable(
+            Path logPath,
+            long currentVersion,
+            AbstractMetadata currentMetadata,
+            AbstractProtocol currentProtocol) {
+        java.util.Map<String, AttributeValue> item = new HashMap<>();
+
+        String tableId = java.util.UUID.randomUUID().toString();
+        item.put(DynamoDBTableEntryConstants.TABLE_ID, new AttributeValue().withS(tableId));
+
+        // We maintain the invariant that a commit will only succeed if the latestVersion stored
+        // in the table is equal to attemptVersion - 1. To maintain this, even though the
+        // filesystem-based commit after register table can fail, we still treat the attemptVersion
+        // at registration as a valid version. It is expected that the Delta client will perform
+        // another registration if the filesystem-based commit fails. When the filesystem-based
+        // commit does fail, this entry will not be associated with any Delta table because the
+        // `tableId` generated above must be committed as part of the filesystem-based commit for
+        // this entry to be associated with a Delta table.
+        long attemptVersion = currentVersion + 1;
+        item.put(
+                DynamoDBTableEntryConstants.TABLE_LATEST_VERSION,
+                new AttributeValue().withN(Long.toString(attemptVersion)));
+
+        item.put(
+                DynamoDBTableEntryConstants.TABLE_PATH,
+                new AttributeValue().withS(logPath.getParent().toString()));
+        item.put(DynamoDBTableEntryConstants.COMMITS, new AttributeValue().withL());
+        item.put(
+                DynamoDBTableEntryConstants.ACCEPTING_COMMITS, new AttributeValue().withBOOL(true));
+        item.put(
+                DynamoDBTableEntryConstants.SCHEMA_VERSION,
+                new AttributeValue().withN(Integer.toString(CLIENT_VERSION)));
+
+        PutItemRequest request = new PutItemRequest()
+                .withTableName(managedCommitsTableName)
+                .withItem(item)
+                .withConditionExpression(
+                        String.format(
+                                "attribute_not_exists(%s)", DynamoDBTableEntryConstants.TABLE_ID));
+        client.putItem(request);
+
+        Tuple2<String, String> tableIdEntry = new Tuple2<>(DynamoDBTableEntryConstants.TABLE_ID, tableId);
+        Map<String, String> tableConf = Map$.MODULE$.<String, String>empty().$plus(tableIdEntry);
+
+        return tableConf;
+    }
+
+    // Copied from DynamoDbLogStore. TODO: add the logging back.
```

Comment:
What logging needs to be added back?

Reply:
1. That the table exists/does not exist
2. Table creation attempt
3. Table creation failure
I will add these and logs in other parts of this class in a follow up PR.